### PR TITLE
Initial setup for webpack-4

### PR DESCRIPTION
**Pull request by GeertWille**
Hey,

Initial setup for webpack 4. I tested some examples and they all seemed to work fine. Please test it out with your projects and post the issues you get here.

Thanks!

### CODE REVIEW

**Comment by denis-sokolov** [comment]
- [ ] A major version number bump. React-static exposes webpack config to the client, and in my case, I rely on Webpack 3 plugins. The upgrade will be breaking backwards compatibility.
- [ ] `extractCssChunks` option relies on the `extract-css-chunks-webpack-plugin` plugin, which is not updated for Webpack 4.
- [ ] Once we are able to migrate to Webpack 4, let’s remember to revisit webpack-serve in #504.

File: package.json [hunk 1]
```diff
@@ -61,12 +59,12 @@
     "extract-css-chunks-webpack-plugin": "^2.0.18",
     "extract-hoc": "^0.0.5",
     "extract-hoc-compose": "^0.0.1",
-    "extract-text-webpack-plugin": "^3.0.1",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
```
Comment (poorpaddy) [inline package.json]: I believe this is no longer being supported or contributed to. See https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/749. 

You may want to follow the pattern in this PR which uses mini-css-extract-plugin instead: https://github.com/phoenixframework/phoenix/pull/2831/files

**Comment by sudarshang** [comment]
How can I help get this PR merged?

**Comment by JustFly1984** [comment]
"extract-hoc": "^0.0.5" and "extract-hoc-compose": "^0.0.1" does not
required by HMR@4, which is supported by react-static.



2018-05-07 23:51 GMT+08:00 Sudarshan Gaikaiwari <notifications@github.com>:

**Comment by sudarshang** [comment]
I have a new PR out #605, that addresses most of the concerns raised here. Please have a look and give feedback.

**Comment by tannerlinsley** [comment]
Going with #605

**Pull request closed without merging**